### ISSUE TITLE
Maven Artifact Repository

### ISSUE THREAD

**Issue by QbDesu** [Azanor/Baubles#266]
It would really be nice if you could publish Baubles to a maven artifact repository server. It would make adding and updating it as a dependency a lot easier.
If you'd like I'd even gladly host it on my own server. It's running Sonatype Nexus Repo OSS. It's configured for https/TLS already and creating a user configuration for you is no problem.
Of course, any server would do, really. I could even help with setting up the build script if you'd like me to.
Pretty please... ? ^_^'

**Comment by Xalcon** [comment]
you could use the curseforge maven. All Projects that are uploaded to curseforge are (in theory) available via their maven. There are a few issues with naming since a lot of projects dont use maven compatible names but baubles is not an issue. A "real" maven server would be nice though since the curse maven is a bit iffy for certain projects.
```java
repositories {
    maven {
        name = "CurseForge"
        url = "https://minecraft.curseforge.com/api/maven/"
    }
}

dependencies {
    compile 'baubles:Baubles:1.12:1.5.2'
}
```

https://authors.curseforge.com/docs/api

**Comment by QbDesu** [comment]
Oh my, I've never seen this before. This is great! Thank you a lot! The only thing I could complain about now would be missing sources but that's really the least of problems.
Btw. I also found this one http://maven.thiakil.com/com/azanor/baubles/Baubles/ in the Botania. Is this in any way official? Sorry but I don't just want to add the maven repo of somebody unrelated to the project.

**Comment by Xalcon** [comment]
i wouldnt consider it to be official, not as long as there is no feedback from azanor. For simple deplyoment to curse this should be fine but we dont know how often this maven is updated with new versions. The curse maven on the other hand will update everytime a new version is uploaded.

**Comment by Daomephsta** [comment]
The CF maven works, but it's really badly set up. I've had the artifact ID change on me literally overnight, and in the same case multiple artifact IDs for different files from the same project (API & full jar). It's also possibly the cause of certain maven deps I have not working with deobfCompile. 

So I would really appreciate an official maven for Baubles, as the CurseForge Maven is terrible.

**Comment by BrainStone** [comment]
I agree. A mod this essential to the modding community should be on a maven!

**Comment by ADDSynth** [comment]
What!? It's not on an official maven yet? I am shocked! No wonder I couldn't figure out which maven I'm supposed to get it from!

**Comment by ADDSynth** [comment]
I'd like to remind everyone that MinecraftForge also has a maven themselves:

[https://files.minecraftforge.net/maven/](https://files.minecraftforge.net/maven/)

If you're interested, you can check it out. But I don't know anything about how to use it to host files there.

**Comment by ADDSynth** [comment]
Possibly on a related note but still posting here regardless. I heard once that the developer of Baubles (Azanor) is no longer developing Minecraft mods. However, as a replacement for Baubles, people can use the [Curios API](https://www.curseforge.com/minecraft/mc-mods/curios) for Minecraft versions 1.13 and up. It has similar functionality. And any mod that is on CurseForge can use CurseForge's [maven API](https://authors.curseforge.com/knowledge-base/projects/529-api) to automatically download the libraries to use in dev projects. As vast as the mod repository CurseForge is, many developers also get their libraries from there.
